src/Kathara/manager/kubernetes/KubernetesMachine.py

```diff
@@ -458,11 +458,17 @@ def _build_definition(self, machine: Machine, config_map: client.V1ConfigMap) ->
                 )
             ))
 
+        # Add image_pull_secrets if using a private registry
+        image_pull_secrets_arg = {}
+        if Setting.get_instance().private_registry_dockerconfigjson:
+            image_pull_secrets_arg = {"image_pull_secrets": [client.V1LocalObjectReference(name="private-registry")]}
+
         pod_spec = client.V1PodSpec(containers=[container_definition],
                                     hostname=machine.meta['real_name'],
                                     dns_policy="None",
                                     dns_config=dns_config,
-                                    volumes=volumes
+                                    volumes=volumes,
+                                    **image_pull_secrets_arg
```

Comment:
You also need to delete the Secret when the lab is undeployed or everything is wiped!

Reply:
When the namespace is deleted all secrets are automatically deleted.
I would still keep the deletion function for the secrets in case someone in the future would want to delete a secret without undeploying the lab.

The wipe function instead is useless, but we could still do something like this for consistency:
```python
def wipe(self) -> None:
    """Namespace deletion implies automatic deletion of all secrets.

    Returns:
        None
    """
    pass
```